libkmod/libkmod.c

```diff
@@ -810,22 +810,37 @@ KMOD_EXPORT void kmod_unload_resources(struct kmod_ctx *ctx)
 
 KMOD_EXPORT int kmod_dump_index(struct kmod_ctx *ctx, enum kmod_index type, int fd)
 {
+	int err = 0, fd2;
+	FILE *fp;
+
 	if (ctx == NULL)
 		return -ENOSYS;
 
+	fd2 = dup(fd);
```

Comment:
AFAICT libraries should really be using CLOEXEC, aka `fcntl(fd, F_DUPFD_CLOEXEC...)` or alike.

Reply:
The `fdopen` would set ´O_CLOEXEC`. Unfortunately I cannot simply use `dup3`, which would do it atomically for us, because then I would have to figure out a free fd first (or open one just to reserve it).